src/Generator/CodeGenerator.php

```diff
@@ -616,9 +616,7 @@ function ($a, $b) use ($imports) {
      */
     private function validateEnumEntity(string $entity_class)
     {
-        $reflector = new \ReflectionClass($entity_class);
-
-        if (! in_array(EnumeratorCompatibleEntityInterface::class, $reflector->getInterfaceNames())) {
+        if (! class_implements($entity_class, EnumeratorCompatibleEntityInterface::class)) {
```

Comment:
Return type is an array (the classes it implements). The signature like this won't work either: 
```
array class_implements ( mixed $class [, bool $autoload = TRUE ] )
```